.buildkite/scripts/run_autoformat.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python3
+# -*- encoding: utf-8
+"""
+This script does autoformatting in Buildkite on pull requests.
+
+In particular, it runs the 'make format' task, and if there are any changes,
+it pushes a new commit to your pull request and aborts the current build.
+"""
+
+import os
+import subprocess
+import sys
+
+
+def branch_name():
+    """Return the name of the branch under test."""
+    return os.environ["BUILDKITE_BRANCH"]
+
+
+def check_call(cmd):
+    """
+    A wrapped version of subprocess.check_call() that doesn't print a
+    traceback if the command errors.
+    """
+    print("*** Running %r" % " ".join(cmd))
+    try:
+        return subprocess.check_call(cmd)
+    except subprocess.CalledProcessError as err:
+        print(err)
+        sys.exit(err.returncode)
+
+
+def git(*args):
+    """Run a Git command and return its output."""
+    cmd = ["git"] + list(args)
+    try:
+        return subprocess.check_output(cmd).decode("utf8").strip()
+    except subprocess.CalledProcessError as err:
+        print(err)
+        sys.exit(err.returncode)
+
+
+def make(*args):
+    """Run a Make command, and check it completes successfully."""
+    check_call(["make"] + list(args))
+
+
+def get_changed_paths(*args):
+    """
+    Returns a set of changed paths in a given commit range.
+
+    :param commit_range: Arguments to pass to ``git diff``.
+    """
+    diff_output = git("diff", "--name-only", *args)
+
+    return set([line.strip() for line in diff_output.splitlines()])
+
+
+if __name__ == "__main__":
+
+    make("format")
+
+    # If there are any changes, push to GitHub immediately and fail the
+    # build.  This will abort the remaining jobs, and trigger a new build
+    # with the reformatted code.
+    if get_changed_paths():
```

Comment:
minor: maybe replace this with a method that returns a boolean `has_uncommitted_changes()`?